pkg/template/values.go

```diff
@@ -102,6 +110,105 @@ func (t Values) writeFromSecret(dstPath string,
 	return result, nil
 }
 
+func (t Values) writeFromDownwardAPI(dstPath string, downwardAPIRef v1alpha1.AppTemplateValuesDownwardAPI) ([]string, error) {
+	var result []string
+
+	for _, item := range downwardAPIRef.Items {
+		content, err := t.extractFieldPathAsKeyValue(item.Name, item.FieldPath)
+		if err != nil {
+			return nil, err
+		}
+
+		//TODO: should path be an allowed config.
+		// cue seems to want the file to have encoding
+		path, err := t.writeFile(dstPath, item.Name+".yaml", content)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, path)
+	}
+
+	sort.Strings(result)
+
+	return result, nil
+}
+
+func (t Values) extractFieldPathAsKeyValue(name string, fieldPath string) ([]byte, error) {
```

Comment:
wdyt about using json path library that kubectl uses for extracting values?

Reply:
looks like it's using https://github.com/kubernetes/client-go/tree/master/util/jsonpath.

i think it's still probably a good idea to limit, but instead of limiting the json path, we can limit input structure. that way you dont have make assertions against a query.